### ISSUE TITLE
Analytics Implementation & QA Support | Language Assistance and Resource Pages | C&L Team

### ISSUE THREAD

**Issue by SKasimow** [department-of-veterans-affairs/va.gov-team#24408]
# Google Analytics Implementation or QA Request

## What this form is for

<details>
<summary>More Information</summary>
<div>
    <br>
<p>Use this template to request Google Tag Manager and Google Analytics implementation or QA for your product. Please fill out all brackets.</p>
    
> _Please see the following information for implementation timeline:_
> New Teams: Implementation will usually be completed in 4 sprints:
> 
> - Sprint 1: Discovery/Orientation
> - Sprint 2: FE Implementation
> - Sprint 3: Tagging Implementation
> - Sprint 4: QA
> 
> All other teams: Implementation will usually be completed in 2 sprints
> 
> - Sprint 1: Discovery and FE Implementation
> - Sprint 2: Tagging Implementation and QA

For more information, please check out the [Google Tag Manager resources](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/analytics/google-analytics/readme.md).

An example of a completed request template can be found [here](https://github.com/department-of-veterans-affairs/va.gov-team/issues/7199). A fully completed issue can help speed up the implementation process.
</div>
</details>

## Description

**Is this request for Google Analytics implementation or QA?**

- The initial request is for GA implementation but will need QA as well.

**Who is this request for?**

 - @SKasimow @DanielleThierryUSDSVA 

    _Provide any relevant contact information and link this issue to the insights team on the DSVA Slack Channel: `#VFS-Platform-Support`._
    
**What team/project is this work for?**

Content & Localization team

1. This is for 3 new Language Assistance and Resource pages (Spanish, Tagalog, Other Languages)
AND
2. Addition of link in the footer to the new pages
    
**Provide the Domain where Google Analytics needs to be implemented**
    
1. staging.va.gov

https://staging.va.gov/asistencia-y-recursos-en-espanol/
https://staging.va.gov/tagalog-wika-mapagkukunan-at-tulong/https://va.gov/asistencia-y-recursos-en-espanol/

 
2. va.gov

https://www.va.gov/resources/how-to-get-free-language-assistance-from-va/

**Identify if PII is passed in URL or title tag**

- N/A

**Please provide any upcoming deadlines or important milestones**

- Launch is targeted for June 7 2021

**Typical Website Interactions to Track**
 
Conversions 
Understanding the User Flow

**Provide URLs that are part of the application from introduction page to successful form completion; conditional pages that are visible depending on an applicant's answer should be marker accordingly.**
    
Final URLs are TBD but will likely be one of the options below (Spanish example below)

 - www.va.gov/spanish-language-assistance-support-and-resources/
- www.va.gov/espanol-language-assistance-support-and-resources/
- www.va.gov/apoyo-y-recursos-en-espanol/ (would need translation validation)

**Other User Interactions: Provide details on other interactions and metrics, like click to help text links, that are helpful in understanding user behavior.**

- Link tracking
- Pageviews
- Scroll depth
- Click to call
- Entry and exit pages
- UVs
- Satisfaction scores
    

**Sample Users when authentication is required _**do not put staging credentials in your `va.gov-team` ticket; store or reference them in a .md file in the `va.gov-team-sensitive` repository**_

    - No authentication required

**Test Users to replicate the event _**do not put staging credentials in your `va.gov-team` ticket; store or reference them in a .md file in the `va.gov-team-sensitive` repository**_

- N/A
    
**Certain conditions/responses that need to be reviewed**

-Can Analytics team provide the event names we should use for the footer, such as:

FOOTER_COLUMNS.PROGRAMS: 'nav-footer-programs
FOOTER_COLUMNS.RESOURCES: 'nav-footer-resources

**Additional Comments**

> Please leave the following blank

## Acceptance Criteria
- [ ] QA has been completed
- [ ] Events have been added to `Product` and `Benefit Hub` Content Groups
- [ ] VFS team completes brief [VSP Collaboration Cycle Feedback](https://adhoc.optimalworkshop.com/questions/20260uu8-0-0/questions/before) survey

## Definition of Done
- [ ] All appropriate issue tagging is completed
- [ ] All AC completed
- [ ] VSP: [Platform Collaboration Point Tracker](https://docs.google.com/spreadsheets/d/1OgPyEvUlNF6EnaYMFAXJkV6FKOvZnlPnbOQ2fAJ7W7A/edit#gid=1116695886) is updated

**Comment by SKasimow** [comment]
FYI - @youngfreezyVA . How soon do you need the event name for footer links? We should finalize the URLs by early next week.

**Comment by jonwehausen** [comment]
@laurensimpson14 & @bsmartin-ep Please triage, assign & review. 

For footer events please see our ^nav-footer.*` should do the trick. 

May also need to add scroll depth tracking within GTM.

**Comment by SKasimow** [comment]
@laurensimpson14 @bsmartin-ep Hello - just checking in to see if you have guidance for this? Thanks!

**Comment by joanneesteban** [comment]
Hi @SKasimow - thanks for the ping! We'll get someone to look at these for your launch :)

**Comment by joanneesteban** [comment]
@laurensimpson14 will be working on this implementation.

**Comment by laurensimpson14** [comment]
Hi @SKasimow do you have the URLs finalized for the three new language pages? When I click on any of the URLs listed in the ticket I get a missing page notice.

**Comment by SKasimow** [comment]
@laurensimpson14 Hi - we are going to use translated URLs and are waiting on translation service for URLs and hope to have today/tomo. Once I have them I will add to the ticket.

@youngfreezyVA:  Jon commented above that for footer events:

> "... please see our ^nav-footer.*` should do the trick.

> May also need to add scroll depth tracking within GTM."

Does that suffice for now?

**Comment by youngfreezyVA** [comment]
yes that works, i'll use the following event:
`nav-footer-language-support` and I can pass the selected lang, similar to the analytics for the language switcher component

**Comment by SKasimow** [comment]
@laurensimpson14 Here are the 3 URLs for the new pages. Hoping to have 1+ 2  in staging by tomorrow:

1. va.gov/asistencia-y-recursos-en-espanol
2. va.gov/tagalog-wika-mapagkukunan-at-tulong
3. https://www.va.gov/resources/how-to-get-free-language-assistance-from-va/

**Comment by SKasimow** [comment]
Hi @laurensimpson14 do you need to just review the staging instances prior to our staging review (which is June 7) to see if there are any non-standard elements we need to tag? And then test to make sure everything fires off ok?

Just want to clarify what we need by next Thursday for Staging!

Thanks,

Sharon

cc @joanneesteban

**Comment by laurensimpson14** [comment]
Hi @SKasimow,
Please see my notes for each tracking item below: 

## - Link Tracking:
I want to confirm with you what you'd like tracked to be sure we're on the same page and capturing everything that you need. 
**What is currently tracking**:
 - All PDF downloads are automatically tracked: 
![image](https://user-images.githubusercontent.com/80417436/120387476-f9e24b80-c2f7-11eb-914c-8b2cf921b88f.png)

 - All outbound link clicks (such as the one pictured below that goes to http://www.publichealth.va.gov) are automatically tracked: 
![image](https://user-images.githubusercontent.com/80417436/120387995-90167180-c2f8-11eb-8cdc-5788f3d63f83.png)

 - The footer links are now tracking: 
![image](https://user-images.githubusercontent.com/80417436/120389450-6f4f1b80-c2fa-11eb-9651-d95b06b171ea.png)

 - There are four links that go to other internal pages: 
          1. https://staging.va.gov/coronavirus-veteran-frequently-asked-questions-tag/
          2. https://staging.va.gov/health-care/covid-19-vaccine-tag/
          3. https://staging.va.gov/health/rights/patientrights.asp
          4. https://staging.va.gov/health/rights/familyrights.asp

     Because these URLs are not repeated anywhere on the page, we recommend viewing the analytics for these by sorting by previous page path, rather than creating a new event. 

 - The "Find helpful VA phone numbers" CTA on the "How to get free language assistance from VA" page is tracking: 
![image](https://user-images.githubusercontent.com/80417436/120392229-331dba00-c2fe-11eb-9e59-09b500ef76c6.png)

- These navigation list links are being tracked: 
![image](https://user-images.githubusercontent.com/80417436/120392675-d1aa1b00-c2fe-11eb-812c-9b40013f7831.png)

**What is not currently tracking**: 
 - Accordion clicks are coming in as not set. @youngfreezyVA can you please pass `nav-accordion-embedded-link-click`? (GA report w/those events for reference [here](https://analytics.google.com/analytics/web/#/report/content-event-events/a50123418w177519031p184624291/explorer-segmentExplorer.segmentId=analytics.eventLabel&explorer-table.plotKeys=%5B%5D&explorer-table.filter=embedded/))
![image](https://user-images.githubusercontent.com/80417436/120393383-d4f1d680-c2ff-11eb-8380-d6cd3fd31720.png)

 - Would you like the three jump links here tracked on the Tagalog and Spanish pages?:
![image](https://user-images.githubusercontent.com/80417436/120387307-b982cd80-c2f7-11eb-9186-98a3fb750088.png)

 - I noticed that the link to the facility locator is in two places on the page: 
![image](https://user-images.githubusercontent.com/80417436/120389051-ecc65c00-c2f9-11eb-81b2-4c940ad98e12.png)
![image](https://user-images.githubusercontent.com/80417436/120389128-010a5900-c2fa-11eb-839f-77cb47a991c0.png)

Will you need to be able to distinguish between the two locations on the page - or is just knowing how many users go from the Tagalog/Spanish page to the facility locator (regardless of which button) sufficient? 

## - Pageviews:
These are automatically tracked.

## - Scroll depth:
This is currently tracking on the "How to get free language assistance from VA" page.
![image](https://user-images.githubusercontent.com/80417436/120392770-f7372480-c2fe-11eb-9b13-c1d18b65e045.png)
However, scroll depth is not currently tracking on va.gov/asistencia-y-recursos-en-espanol or va.gov/tagalog-wika-mapagkukunan-at-tulong/. I've added tracking for it into GTM. After this gets QA'd by another member of the analytics team I can publish it. 

## - Click to call:
This is automatically being tracked: 
![image](https://user-images.githubusercontent.com/80417436/120391688-66137e00-c2fd-11eb-94af-57f4c19593b6.png)

## - Entry and exit pages:
These are automatically tracked.

## - UVs:
Do you mean unique visitors? If so, these are automatically tracked.

## - Satisfaction Score:
Are you referring to the "good" and "bad" radio buttons under "How do you rate your experience on this page?"? If so, this is being tracked:
![image](https://user-images.githubusercontent.com/80417436/120392459-855edb00-c2fe-11eb-9cf2-0766fd22a458.png)

## - Other
 - The search bar at the top of the "How to get free language assistance from VA" page is not returning any results, so the only tracking I'm able to verify for this is the `No Search Results Returned` event:
![image](https://user-images.githubusercontent.com/80417436/120392060-f2be3c00-c2fd-11eb-8348-26cba2971447.png)

**Comment by SKasimow** [comment]
Hi Lauren,

Thanks for your detailed view and screenshots!  I just had a few comments/questions on the info you provided above:

## Item 1
> There are four links that go to other internal pages:
> 1. https://staging.va.gov/coronavirus-veteran-frequently-asked-questions-tag/
> 2. https://staging.va.gov/health-care/covid-19-vaccine-tag/
> 3. https://staging.va.gov/health/rights/patientrights.asp
> 4. https://staging.va.gov/health/rights/familyrights.asp
> 
> Because these URLs are not repeated anywhere on the page, we recommend viewing the analytics for these by sorting by previous page path, rather than creating a new event.

**SK: do you know why the other links you reviewed are all tracking and these 4 aren't? Is it bc the others go to teammate (external) pages whereas these go to internal?** 
What is not currently tracking:

## Item 2
> Accordion clicks are coming in as not set. @youngfreezyVA can you please pass nav-accordion-embedded-link-click? (GA report w/those events for reference here)
> image

**SK: need @youngfreezyVA to fix this**

## Item 3 
> Would you like the three jump links here tracked on the Tagalog and Spanish pages?:
> image

**SK: yes we should track these; is that something @youngfreezyVA needs to add?**

## Item 4

> I noticed that the link to the facility locator is in two places on the page:
> image

SK: I think this is just placeholder text but will ask @DanielleThierryUSDSVA to review

To be continued in next comment....

**Comment by SKasimow** [comment]
## Item 5

> Other
> 
> The search bar at the top of the "How to get free language assistance from VA" page is not returning any results, so the only tracking I'm able to verify for this is the No Search Results Returned event:
> 

SK: think this is just an artifact of staging set-up. @youngfreezyVA can you confirm??

Thanks All!

**Comment by laurensimpson14** [comment]
> SK: do you know why the other links you reviewed are all tracking and these 4 aren't? Is it bc the others go to teammate (external) pages whereas these go to internal?

LS: Yes, we automatically track links to external pages since it is difficult to see GA data across multiple domains and multiple GA properties. For example, with external pages, we can't go the pageviews report, filter by that page, and then add a secondary dimension of `previous page path` since it can't look back across domains to see the previous page. Instead, it will register the external page as the landing page/first page visited in a different GA property. 

With internal pages (like the 4 in the list) we don't typically add event tracking for those since you can find the information you're looking for in the pageviews report. This is also an effort to reduce the number of hits we send to GA since there is a limit for hits collected. 

> LS: Would you like the three jump links here tracked on the Tagalog and Spanish pages?:
image
> SK: yes we should track these; is that something @youngfreezyVA needs to add?

LS: Yes - data layer specification for jump links: `'event': 'nav-jumplink-click'`, add link name attribute (or similar)

**Comment by SKasimow** [comment]
@youngfreezyVA Hi Fareez - think this is the only addition we need to make for GA tracking:


For "On this page" add tracking for the three jump links on the Tagalog and Spanish pages.

The data layer specification for jump links: 'event': 'nav-jumplink-click', add link name attribute (or similar)

![image.png](https://images.zenhubusercontent.com/5fb57aafa11eae0592a15cb1/3ed3f4e5-3d53-45ca-b4fe-905b53709783)

**Comment by youngfreezyVA** [comment]
awesome! thanks so much @SKasimow for grooming this analytics ticket 🎉 🎖️
I will include these changes in the PR that updates Language Support -> Language Assistance

**Comment by youngfreezyVA** [comment]
after doing some digging, it looks like the above use case is already covered:
https://github.com/department-of-veterans-affairs/content-build/blob/master/src/site/stages/build/plugins/modify-dom/add-id-to-subheadings.js#L93
![Uploading Screen Shot 2021-06-02 at 10.27.43 AM.png…]()

**Comment by laurensimpson14** [comment]
@youngfreezyVA I don't see any events firing for the table of contents. I'm also not seeing it in GitHub: https://github.com/department-of-veterans-affairs/va.gov-team/search?q=nav-table-of-contents

Can I ask where that came from?

**Comment by youngfreezyVA** [comment]
hello @laurensimpson14 ... the link above is where the table of contents is referenced, please take a look at this file or review internally with the analytics team and they can help clarify:
https://github.com/department-of-veterans-affairs/content-build/blob/master/src/site/stages/build/plugins/modify-dom/add-id-to-subheadings.js#L84

**Comment by laurensimpson14** [comment]
@youngfreezyVA Based on the code you linked, we should be seeing `nav-jumplink-click` events on table of contents link click? However, I'm not seeing them on the staging link.

**Comment by youngfreezyVA** [comment]
<img width="962" alt="Screen Shot 2021-06-03 at 7 48 01 AM" src="https://user-images.githubusercontent.com/46791771/120668169-0de69400-c443-11eb-8652-d29b853a2f14.png">
it lo
![Uploading Screen Shot 2021-06-02 at 10.35.34 AM.png…]()
it looks like there is an issue with the inline recordEvent from the above code
fixing this would require a separate ticket as I did not make the original changes
I am guessing this may be an issue specific to the staging environment, but at this moment I'm not 100%

**Comment by youngfreezyVA** [comment]
@bsmartin-ep has previously reported errors from this inline event, he may be able to provide more insight.

**Comment by laurensimpson14** [comment]
I spoke with @bsmartin-ep and he's going to create a ticket for that error. 

In the meantime we can QA this one item when it goes to production @SKasimow.

**Comment by SKasimow** [comment]
@laurensimpson14 sounds good, thanks Lauren!

**Comment by joanneesteban** [comment]
@SKasimow let us know when this is pushed to Production, and we can double check that this is good to go! (CC: @bsmartin-ep )

**Comment by SKasimow** [comment]
@joanneesteban Hi - this is now first full day live (June 11th)!

**Comment by bsmartin-ep** [comment]
As expected, table of content links at the top of the page are working correctly in production.

![image](https://user-images.githubusercontent.com/62304138/121760316-1486a280-caf8-11eb-82af-926251e11877.png)

![image](https://user-images.githubusercontent.com/62304138/121760309-0cc6fe00-caf8-11eb-9bb1-42e749476c0f.png)

**Comment by joanneesteban** [comment]
@SKasimow congrats on the launch! Can we close this ticket?

**Comment by SKasimow** [comment]
@joanneesteban Hi - thanks! I did check and am seeing data for page views (the only thing I checked so far) but do you guys check the other key metrics once it is moved to production? I know Brian confirmed the "On this page" component for us as noted above.

If everything is verified on you end then yes, feel free to close the ticket. Thanks, Sharon

**Comment by laurensimpson14** [comment]
Hi @SKasimow, I've gone through and completed QA for each component in production below! Please see my notes and let me know if there is anything missing (note: entry & exit pages as well as unique visitors vs returning visitors are automatically tracked with pageviews) 🙂 

| Interaction | Screenshot | Status |
| :------- | :-------: | :-------:|
|PDF Downloads - | ![image](https://user-images.githubusercontent.com/80417436/122802491-67e4b780-d293-11eb-84af-4cb582bb9b04.png) | 🟢 Tracking Correctly|
| Outbound Link Clicks | ![image](https://user-images.githubusercontent.com/80417436/122802775-c14ce680-d293-11eb-8c2b-44295926bfec.png) | 🟢 Tracking Correctly|
| Footer Language Assistance Links | ![image](https://user-images.githubusercontent.com/80417436/122802850-e17ca580-d293-11eb-8d4c-ac6fdf61df78.png)| 🟢 Tracking Correctly|
|`Find helpful VA phone numbers` CTA | ![image](https://user-images.githubusercontent.com/80417436/122803006-1a1c7f00-d294-11eb-8f4b-cc6b68d5bbf2.png)| 🟢 Tracking Correctly|
|Navigation List Links| ![image](https://user-images.githubusercontent.com/80417436/122803116-34eef380-d294-11eb-9bb6-32dce059f1ee.png)| 🟢 Tracking Correctly|
|Accordion clicks |![image](https://user-images.githubusercontent.com/80417436/122803283-6667bf00-d294-11eb-927e-396c6d5c815d.png)| 🟡  Accordion clicks are coming in as not set. @youngfreezyVA can you please pass nav-accordion-embedded-link-click? (GA report w/those events for reference [here](https://analytics.google.com/analytics/web/#/report/content-event-events/a50123418w177519031p184624291/explorer-segmentExplorer.segmentId=analytics.eventLabel&explorer-table.plotKeys=%5B%5D&explorer-table.filter=embedded/))|
|Jump Links | ![image](https://user-images.githubusercontent.com/80417436/122803370-84352400-d294-11eb-9ae2-de11a99553e3.png)| 🟢 Tracking Correctly|
|Pageviews|![image](https://user-images.githubusercontent.com/80417436/122803427-9911b780-d294-11eb-8c4b-5c5673f58c76.png)| 🟢 Tracking Correctly|
|Scroll Depth| ![image](https://user-images.githubusercontent.com/80417436/122803492-af1f7800-d294-11eb-804e-855c54014dcb.png)| 🟢 Tracking Correctly|
|Click to Call| ![image](https://user-images.githubusercontent.com/80417436/122803589-cb231980-d294-11eb-8472-241e12f73688.png)| 🟢 Tracking Correctly|
|Satisfaction Score (Good/Bad Buttons)| ![image](https://user-images.githubusercontent.com/80417436/122816305-8c955b00-d2a4-11eb-8733-479f3299de52.png)| 🟡 the `not set` portion of the interaction label should be `good` or `bad` depending on the selection, not `not-set` @youngfreezyVA do you know why this is happening? (link to page [here](https://www.va.gov/resources/how-to-get-free-language-assistance-from-va/))|
|Search Bar| ![image](https://user-images.githubusercontent.com/80417436/122816988-6328ff00-d2a5-11eb-95df-0b0e1252448d.png)| 🟢 Tracking Correctly|

**Comment by SKasimow** [comment]
@laurensimpson14 Thanks so much Lauren! Will look into the issues you noted above

**Comment by SKasimow** [comment]
@laurensimpson14  Closing this issue and opened the following ticket to fix 2 items noted in comment above.
https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/26434
